crypto/hashes.go

```diff
@@ -90,3 +109,32 @@ func HashBytes(hash hash.Hash, m []byte) []byte {
 	outhash := hash.Sum(nil)
 	return outhash
 }
+
+// InvalidHash is used to identify errors on the factory.
+// this function will return nil slice
+type InvalidHash struct {
```

Comment:
make `InvalidHash` into a small case `invalidHash` since there is no point where you would export this object.